lib/src/main/scala/spinal/lib/bus/regif/BusIf.scala

```diff
@@ -0,0 +1,292 @@
+    }
+  }
+
+  private def attachAddr(sizemap: SizeMapping) = {
+    val ret = regAddressMap.filter(_.overlap(sizemap))
+    val t = regAddressHistory.filter(t => (t <= sizemap.end) &&  (t >= sizemap.base))
+    if(!ret.isEmpty){
+      SpinalError(s"${ret.head} overlap with ${sizemap}")
+    } else if(!t.isEmpty){
+      SpinalError(s"0x${t.head.hexString()} overlap with ${sizemap}")
+    } else {
+      regAddressMap.append(sizemap)
+    }
+  }
+
+  def getRegPtr(): BigInt = regPtr
+
+  /*Attention: Should user make address no conflict them selves*/
+  def regPtrReAnchorAt(pos: BigInt) = {
+    require(pos % (busDataWidth/8) ==0, s"Address Postion need allign datawidth ${busDataWidth/8} byte")
+    regPtr = pos
+  }
+
+  private def checkLastNA(): Unit = SliceInsts.foreach(_.checkLast)
+  private def regNameUpdate(): Unit = {
+    val words = "\\w*".r
+    val pre = regPre match{
+      case "" => ""
+      case words(_*) => regPre + "_"
+      case _ => SpinalError(s"${regPre} should be Valid naming : '[A-Za-z0-9_]+'")
+    }
+    RegInsts.foreach(t => t.setName(s"${pre}${t.getName()}"))
+  }
+
+  private var isChecked: Boolean = false
+  def preCheck(): Unit = {
+    if(!isChecked){
+      checkLastNA()
+      regNameUpdate()
+      isChecked = true
+    }
+  }
+
+  component.addPrePopTask(() => {
+    this.readGenerator()
+  })
+
+
+  def regPart(name: String)(block : => Unit) = {
+    this.newBlockTag(name)(name)
+    block
+    this.resetBlockTag()
+  }
+
+
+  def newRegAt(address: BigInt, doc: String, grp: GrpTag = null)(implicit symbol: SymbolName) = {
+    assert(address % wordAddressInc == 0, s"located Position not align by wordAddressInc: ${wordAddressInc}")
+    val reg = creatReg(symbol.name, address, doc, grp)
+    regPtr = address + wordAddressInc
+    reg
+  }
+
+  def newReg(doc: String, grp: GrpTag = null)(implicit symbol: SymbolName) = {
+    val res = creatReg(symbol.name.toLowerCase(), regPtr, doc, grp)
+    regPtr += wordAddressInc
+    res
+  }
+
+  def creatReg(name: String, addr: BigInt, doc: String, grp: GrpTag = null) = {
+    val ret = new RegInst(name, addr, doc, this, grp)
+    SliceInsts += ret
+    attachAddr(regPtr)
+    ret
+  }
+
+  def newRAM(size: BigInt, doc: String, grp: GrpTag = null)(implicit symbol: SymbolName) = {
+    val res = creatRAM(symbol.name.toLowerCase(), regPtr, size, doc, grp)
+    regPtr += scala.math.ceil(size.toDouble/wordAddressInc).toLong * wordAddressInc
+    res
+  }
+
+  def newRAMAt(address: BigInt, size: BigInt, doc: String, grp: GrpTag = null)(implicit symbol: SymbolName) = {
+    assert(address % wordAddressInc == 0, s"located Position not align by wordAddressInc: ${wordAddressInc}")
+    val res = creatRAM(symbol.name, address, size, doc, grp)
+    regPtr = address + scala.math.ceil(size.toDouble/wordAddressInc).toLong * wordAddressInc
+    res
+  }
+
+  def creatRAM(name: String, addr: BigInt, size: BigInt, doc: String, grp: GrpTag = null) = {
```

Comment:
why leave out a single character in `create`? typing this I always have to remind myself to make this mistake.
I see that `creatReg` existed for the old API - we should deprecate that one and fix the spelling.

Reply:
XD，  fixed https://github.com/SpinalHDL/SpinalHDL/pull/1434/commits/e153944f85917791e882c727a25df809b9621a15